### ISSUE TITLE
Can we build compton without a2x?

### ISSUE THREAD

**Issue by jrcharney** [chjj/compton#262]
I'm eager to see this project get a CMakeList file or a configure.sh script set up.  (Since your halfway there on making a `CMakeList.txt` file, the `cmake` route may be better.)

More importantly, compton will not install without running `a2x` and  `a2x` is part of the `asciidoc` package on distros like Debian and Ubuntu.  `asciidoc` wants to install about 1GB of extra stuff , most of which I installed from source rather than older binaries.  This includes Ruby which I installed from source using the RVM.io script and a whole bunch of LaTeX and PostScript related software that I'm not going to use most of the time.

Thus, I would just like to install compton but not the documentation especially since the documentation wants to suck up space on my small disk that I'm using.  (Seeing that `Doxyfile` as in Doxygen is kind of a red flag there.  That's over a gig of unnecessary software I'll never use either.)

The last three lines of output when I run `sudo make install` are

```
a2x --format manpage man/compton.1.asciidoc
make:  a2x: command not found
make: *** [man/compton.1] Error 127
```

TLDR: Make installing compton independent from installing the compton documentation.  I'm looking to install just a 2MB piece of software, not the 1GB wiki entourage that comes with it.

Thank you.

**Comment by WorMzy** [comment]
In the interim,

```bash
sed -i '/MANPAGES/d' Makefile
```

Works for me.

I'd file a bug against the package in your distribution though. Asciidoc should only take up ~800KiB by itself, and on a barebones Arch system, asciidoc+deps only takes up 97MiB. I don't think it would be unreasonable to ask your distro maintainers to provide a asciidoc-minimal package that simply provides the basic functionality. You're only using it for a second to generate a manpage, after all. After that, you can remove it from your system.

**Comment by richardgv** [comment]
Sorry for the late reply, firstly.

> I'm eager to see this project get a CMakeList file or a configure.sh script set up.  (Since your halfway there on making a `CMakeList.txt` file, the `cmake` route may be better.)

I think the current build system with GNU make works alright, and I don't see a point of adding an extra dependency.

> Make installing compton independent from installing the compton documentation.

With 64b4fe2 (`richardgv-dev` branch right now), you will be able to `make install` without documentation, by `make MANPAGES= install`.

If using the sources from git is not applicable, there are other workarounds available:

1. `make`, then `cp compton /usr/bin`. Compton is merely a ELF executable file.

2. Patch the `Makefile`, as what WorMzy suggested.

3. Pre-build the man pages into the source package, or fetch the man pages from binary packages.

> More importantly, compton will not install without running `a2x` and  `a2x` is part of the `asciidoc` package on distros like Debian and Ubuntu.  `asciidoc` wants to install about 1GB of extra stuff , most of which I installed from source rather than older binaries.  This includes Ruby which I installed from source using the RVM.io script and a whole bunch of LaTeX and PostScript related software that I'm not going to use most of the time.

As WorMzy stated, that's a problem of the packager of asciidoc of the distribution you use. But as far as know, if you choose to ignore "suggests" dependencies when you install asciidoc and its dependencies, things may get a lot more better. (TexLive is a "suggests" dependency of dblatex, upon which asciidoc depends, in Debian.)

> (Seeing that `Doxyfile` as in Doxygen is kind of a red flag there.  That's over a gig of unnecessary software I'll never use either.)

Sorry, I don't really understand it. The code documentation generated with Doxygen and `Doxyfile` is used by developers to make comprehending the structure of the project easier, and I don't think casual users would need to build it.
